File: Baekjoon/정렬/5052.py
# https://www.acmicpc.net/problem/5052

import sys
input = sys.stdin.readline

for _ in range(int(input())):
    n = int(input())
    phone = []
    for _ in range(n):
        phone.append(input().rstrip())
    phone.sort()
    flag = True
    for i in range(n - 1):
        if phone[i] == phone[i + 1][:len(phone[i])]:
            flag = False
            break
    if flag:
        print("YES")
    else:
        print("NO")

File: Baekjoon/정렬/5052.cpp
// https://www.acmicpc.net/problem/5052

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

bool is_consistent(vector<string>& a) {
    for (int i = 0; i < a.size() - 1; i++) {
        if (a[i] == a[i + 1].substr(0, a[i].size())) {
            return false;
        }
    }
    return true;
}

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        vector<string> a;

        cin >> n;
        a.resize(n);
        for (int i = 0; i < n; i++) {
            cin >> a[i];
        }

        sort(a.begin(), a.end());
        if (is_consistent(a)) {
            cout << "YES" << endl;
        } else {
            cout << "NO" << endl;
        }
    }

    return 0;
}